### PR TITLE
Implement test procedure image views

### PR DESCRIPTION
**Pull request by allan-allay**
See details in -> #60

### CODE REVIEW

**Comment by vincent-karuri** [comment]
image

**Comment by vincent-karuri** [comment]
image

**Comment by vincent-karuri** [comment]
image

**Comment by vincent-karuri** [comment]
Can't get the images to load

**Comment by vincent-karuri** [comment]
```
06-20 18:38:33.594 32519-32519/io.ona.rdt_app E/JsonFormInteractor: Exception occurred in making view : Exception is : Attempt to invoke interface method 'java.util.List com.vijay.jsonwizard.interfaces.FormWidgetFactory.getViewsFromJson(java.lang.String, android.content.Context, com.vijay.jsonwizard.fragments.JsonFormFragment, org.json.JSONObject, com.vijay.jsonwizard.interfaces.CommonListener, boolean)' on a null object reference
06-20 18:38:33.594 32519-32519/io.ona.rdt_app W/System.err: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.List com.vijay.jsonwizard.interfaces.FormWidgetFactory.getViewsFromJson(java.lang.String, android.content.Context, com.vijay.jsonwizard.fragments.JsonFormFragment, org.json.JSONObject, com.vijay.jsonwizard.interfaces.CommonListener, boolean)' on a null object reference
        at com.vijay.jsonwizard.interactors.JsonFormInteractor.fetchViews(JsonFormInteractor.java:158)
        at com.vijay.jsonwizard.interactors.JsonFormInteractor.fetchFields(JsonFormInteractor.java:144)
        at com.vijay.jsonwizard.interactors.JsonFormInteractor.fetchFormElements(JsonFormInteractor.java:98)
        at com.vijay.jsonwizard.presenters.JsonFormFragmentPresenter.addFormElements(JsonFormFragmentPresenter.java:187)
        at com.vijay.jsonwizard.fragments.JsonFormFragment.onViewCreated(JsonFormFragment.java:186)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1471)
        at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1784)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1852)
        at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:802)
        at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2625)
        at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2411)
        at android.support.v4.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2366)
        at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2273)
        at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:733)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:7325)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

**Comment by vincent-karuri** [comment]
Also implement the buttons that navigate between the screens as in the mock-ups.

**Comment by vincent-karuri** [comment]
Add `"display_back_button": "true"` to the image view pages to display the cancel button on the app bar.

**Comment by allan-allay** [comment]
> Also implement the buttons that navigate between the screens as in the mock-ups.

Done :heavy_check_mark:

**Comment by allan-allay** [comment]
> Add `"display_back_button": "true"` to the image view pages to display the cancel button on the app bar.

Done :heavy_check_mark:

**Comment by allan-allay** [comment]
> Resolve conflicts and update native forms version.

Done :heavy_check_mark:

File: rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java [hunk 1]
```diff
@@ -45,16 +44,12 @@ public void onClick(View v) {
         rootView.findViewById(com.vijay.jsonwizard.R.id.next_button).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Object isSubmit = v.getTag(R.id.submit);
-                if (isSubmit != null && Boolean.valueOf(isSubmit.toString())) {
```
Comment (vincent-karuri) [inline rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java]: This logic still makes sense, please revert it.

It should save if submit tag exists but move to next if it doesn't.

We are modifying the logic for when the rdt test form is launched, based on answers on the registration form.

**Comment by vincent-karuri** [comment]
Images should be added between current steps 3 and 4.

File: rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java (hunk 1 above)
Comment (allan-allay) [reply rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java]: Yeah, I merged whatever is on **master**. To move to next we'll need an additional next in the else of the `if(isSubmit != null)`

**Comment by allan-allay** [comment]
> Images should be added between current steps 3 and 4.

I see 2 screens instead of 3 with the images that are specified in the marvel prototypes :thinking:

File: rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java [hunk 2]
```diff
@@ -44,13 +45,20 @@ public void onClick(View v) {
...
+                Object isSubmit = v.getTag(R.id.submit);
+                if (isSubmit != null && Boolean.valueOf(isSubmit.toString())) {
+                    boolean isSaved = save(false);
+                    if (isSaved) {
+                        try {
+                            jsonFormUtils.launchForm(getActivity(), RDT_TEST_FORM);
+                        } catch (JSONException e) {
+                            Log.e(TAG, e.getStackTrace().toString());
+                        }
+                    } else {
```
Comment (vincent-karuri) [inline rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java]: Why do we need this next here?

It's already part of the outer if-else block.

The next button is either of type submit or simply type next.

**Pull request merged**